client/src/game/gameObjects/gameObjectWorldData.ts

```diff
@@ -17,7 +17,7 @@ class GameObjectWorldData implements IGameObjectWorldData {
 		this.isDirty = {}
 		this.position = { x: 0, y: 0, z: 0 }
 		this.rotation = { x: 0, y: 0, z: 0 }
-		this.destination = this.position
+		this.destination = { x: 0, y: 0, z: 0 }
```

Comment:
What's the idea behind this change?

Reply:
This is to avoid a bug. This assignment does not cause a value copy (as you might expect), but a copy of reference. This means that if you set x on the destination you are also setting it on the position.

Reply:
Yea, understood. It's not a big deal either way. Did you try spreading or object.assign? not sure if a deep copy is necessary

Reply:
Assigning the position likes this is probably more performant than applying a method on an existing obj anyway